server/ControlPlane/Database/Repository.cs

```diff
@@ -453,22 +453,41 @@ UPDATE runs
         }, cancellationToken);
     }
 
-    public async Task UpdateRunAsFinal(long id, CancellationToken cancellationToken)
+    public async Task<(DateTimeOffset createdAt, DateTimeOffset? startedAt, DateTimeOffset? finishedAt)> UpdateRunAsFinal(long id, CancellationToken cancellationToken)
     {
-        await _resiliencePipeline.ExecuteAsync(async cancellationToken =>
+        return await _resiliencePipeline.ExecuteAsync(async cancellationToken =>
         {
             await using var conn = await _dataSource.OpenConnectionAsync(cancellationToken);
             await using var command = new NpgsqlCommand("""
                 UPDATE runs
                 SET final = true
                 WHERE id = $1
+                RETURNING created_at, (run->>'startedAt')::timestamptz, (run->>'finishedAt')::timestamptz
```

Comment:
Updated this per discussions on previous PR #246. @johnstairs it would be good if you could check my handywork here. I am pulling the time stamps out as part of the update query, however, the startedAt and finishedAt timestamps are not in the main table but in the jsonb run field and I am not sure if this extraction is actually more efficient than just doing a GetRun after it is done.